### ISSUE TITLE
bug(menuitem - accessibility): Role menuitem is not announced by screen readers

### ISSUE THREAD

**Issue by DhanyaGopinathTR** [angular/components#28306]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Even though the menuitem role is added by default to the mat-menu-item, this is not reading by the screen readers like NVDA.

For eg:- https://www.geeksforgeeks.org/mat-menu-in-angular-material/

![image](https://github.com/angular/components/assets/86048403/ef861c4c-197e-480e-8538-485f3774458d)

![example-mat-menu-item](https://github.com/angular/components/assets/86048403/9b55bc70-03be-4fdd-b247-a7925c967e1d)

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. Turn on NVDA
2. Click/tab on the menu item
3. Navigate through the menu items using up/down keyboard arrows

### Expected Behavior

The role menuitem should be announced by the screenreaders as it is present in the html while inspecting

### Actual Behavior

Even though role is present in html, it is not announced by screen readers

### Environment

The issue can be reproduced in angular material sites also. Below are my project configurations in the local machine
- Angular: ~13.0.1
- CDK/Material: ~13.0.0
- Browser(s): Chrome, Edge
- Operating System (e.g. Windows, macOS, Ubuntu): Windows
